src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,480 @@
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.Telemetry;
+using Microsoft.CodeAnalysis.MSBuild;
+using File = System.IO.File;
+using Path = System.IO.Path;
+
+namespace AnalyzerRunner
+{
+    /// <summary>
+    /// AnalyzerRunner is a tool that will analyze a solution, find diagnostics in it and will print out the number of
+    /// diagnostics it could find. This is useful to easily test performance without having the overhead of visual
+    /// studio running.
+    /// </summary>
+    class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var options = Options.Create(args);
+            if (options == null)
+            {
+                return;
+            }
+
+            CancellationTokenSource cts = new CancellationTokenSource();
+            Console.CancelKeyPress +=
+                (sender, e) =>
+                {
+                    e.Cancel = true;
+                    cts.Cancel();
+                };
+
+            var analyzerAssemblies = GetAnalyzerAssemblies(options.AnalyzerPath);
+            var analyzers = GetAllAnalyzers(analyzerAssemblies);
+            analyzers = FilterAnalyzers(analyzers, options).ToImmutableArray();
+
+            if (analyzers.Length == 0)
+            {
+                Utilities.WriteLine("No analyzers found", ConsoleColor.Red);
+                Utilities.PrintHelp();
+                return;
+            }
+            var cancellationToken = cts.Token;
+
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            using (MSBuildWorkspace workspace = MSBuildWorkspace.Create())
+            {
+                Solution solution = await workspace.OpenSolutionAsync(options.SolutionPath, cancellationToken).ConfigureAwait(false);
+                bool proceedLookingForAnalyzerReferences = true;
+                while (proceedLookingForAnalyzerReferences)
+                {
+                    proceedLookingForAnalyzerReferences = false;
+                    foreach (var project in solution.Projects)
+                    {
+                        if (project.AnalyzerReferences.Any())
+                        {
+                            proceedLookingForAnalyzerReferences = true;
+                            solution = project.WithAnalyzerReferences(ImmutableArray<AnalyzerReference>.Empty).Solution;
+                        }
+                    }
+                }
+
+                Console.WriteLine($"Loaded solution in {stopwatch.ElapsedMilliseconds}ms");
+
+                if (options.Stats)
+                {
+                    List<Project> csharpProjects = solution.Projects.Where(i => i.Language == LanguageNames.CSharp).ToList();
```

Comment:
curious, why only csharp project?

Reply:
nice that we have this, but we probably want semantic info as well since even if msworkspace failed to open solution properly, most likely all syntax is there, just semantic will be the one all broken.